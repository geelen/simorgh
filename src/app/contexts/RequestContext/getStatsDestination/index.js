/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
