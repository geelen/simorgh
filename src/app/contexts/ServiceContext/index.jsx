import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
