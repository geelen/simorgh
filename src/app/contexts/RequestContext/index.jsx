import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
