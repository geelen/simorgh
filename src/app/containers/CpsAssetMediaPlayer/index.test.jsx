import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import {
  shouldMatchSnapshot,
  isNull,
  suppressPropWarnings,
} from '@bbc/psammead-test-helpers';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';
import CpsAssetMediaPlayerContainer from '.';
import {
  validAresMediaVideoBlock,
  defaultToggles,
} from '../MediaPlayer/fixtureData';

const GenerateMediaPlayer = ({
  /* eslint-disable react/prop-types */
  platform,
  blocks,
  assetUri,
  /* eslint-enable react/prop-types */
}) => (
  <RequestContextProvider
    isAmp={platform === 'amp'}
    service="news"
    statusCode={200}
    platform={platform}
    id="foo"
    pageType="MAP"
    pathname="/pathname"
  >
    <ServiceContextProvider service="news">
      <ToggleContext.Provider
        value={{ toggleState: defaultToggles, toggleDispatch: jest.fn() }}
      >
        <BrowserRouter>
          <CpsAssetMediaPlayerContainer blocks={blocks} assetUri={assetUri} />
        </BrowserRouter>
      </ToggleContext.Provider>
    </ServiceContextProvider>
  </RequestContextProvider>
);

describe('MediaPlayer', () => {
  shouldMatchSnapshot(
    'render the canonical player without a placeholder',
    <GenerateMediaPlayer
      platform="canonical"
      blocks={[validAresMediaVideoBlock]}
      assetUri="/pidgin/123456789"
    />,
  );

  shouldMatchSnapshot(
    'render the amp player',
    <GenerateMediaPlayer
      platform="amp"
      blocks={[validAresMediaVideoBlock]}
      assetUri="/pidgin/123456789"
    />,
  );

  suppressPropWarnings(['assetUri']);
  isNull(
    'is Null when assetUri is not provided',
    <GenerateMediaPlayer
      platform="canonical"
      blocks={[validAresMediaVideoBlock]}
    />,
  );
});
