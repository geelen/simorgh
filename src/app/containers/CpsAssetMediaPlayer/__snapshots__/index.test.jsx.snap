// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer render the amp player 1`] = `
<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":"PT3M11S","thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":"2018-10-22T14:35:32.000Z","embedURL":"https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB/amp"}
    </script>
    <script
      async="true"
      custom-element="amp-iframe"
      src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"
    />
  </head>
  <body>
    .c1 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.c2 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin-top: 1rem;
}

.c0 figure {
  padding-bottom: 0;
}

@media (max-width:63rem) {
  .c0 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 6;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 12;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:63rem) {
  .c0 {
    padding-top: 0.5rem;
    margin-top: 2rem;
  }
}

<div>
      <div
        class="c0"
      >
        <figure
          class="c1"
        >
          <div
            class="c2"
          >
            <amp-iframe
              allowfullscreen="allowfullscreen"
              frameborder="0"
              layout="fill"
              sandbox="allow-scripts allow-same-origin"
              scrolling="no"
              src="https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB/amp"
              title="Media player"
            >
              <amp-img
                alt=""
                attribution=""
                height="16"
                layout="fill"
                placeholder="true"
                src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01k6mtv.jpg"
                srcset="https://ichef.test.bbci.co.uk/images/ic/240xn/p01k6mtv.jpg 240w, https://ichef.test.bbci.co.uk/images/ic/320xn/p01k6mtv.jpg 320w, https://ichef.test.bbci.co.uk/images/ic/480xn/p01k6mtv.jpg 480w, https://ichef.test.bbci.co.uk/images/ic/624xn/p01k6mtv.jpg 624w, https://ichef.test.bbci.co.uk/images/ic/800xn/p01k6mtv.jpg 800w"
                width="9"
              />
            </amp-iframe>
          </div>
        </figure>
      </div>
    </div>
  </body>
</html>
`;

exports[`MediaPlayer render the canonical player without a placeholder 1`] = `
.c1 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.c2 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.c0 {
  margin-top: 1rem;
}

.c0 figure {
  padding-bottom: 0;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border: 0;
  overflow: hidden;
}

@media (max-width:63rem) {
  .c0 {
    grid-column: 1 / span 6;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-column: 3 / span 6;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-column: 6 / span 12;
  }
}

@media (max-width:25rem) {
  .c0 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c0 {
    padding: 0 1rem;
  }
}

@media (min-width:63rem) {
  .c0 {
    padding-top: 0.5rem;
    margin-top: 2rem;
  }
}

<html>
  <head>
    <script
      type="application/ld+json"
    >
      {"@context":"http://schema.org","@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":"PT3M11S","thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":"2018-10-22T14:35:32.000Z","embedURL":"https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB"}
    </script>
  </head>
  <body>
    <div>
      <div
        class="c0"
      >
        <figure
          class="c1"
        >
          <div
            class="c2"
          >
            <iframe
              allow="autoplay; fullscreen"
              allowfullscreen=""
              class="c3"
              scrolling="no"
              src="https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/123456789/p01k6msp/en-GB"
              title="Media player"
            />
          </div>
        </figure>
      </div>
    </div>
  </body>
</html>
`;
